cmd/minishift/cmd/delete.go

```diff
@@ -46,23 +48,41 @@ func runDelete(cmd *cobra.Command, args []string) {
 
 	util.ExitIfUndefined(api, constants.MachineName)
 
+	if err := cluster.UnRegister(api); err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error unregistering the VM: %s", err.Error()))
+	}
+
+	if minishiftConfig.InstanceConfig.IsRegister {
```

Comment:
Does this not come to late now? You already made the unregister call. What if the user selects 'n' now?